gigareview/console/models/ManuscriptsWorker.php

```diff
@@ -16,7 +17,48 @@ public function execute($queue)
     {
 
     }
-    
+
+    /**
+     * @param string $manuscriptPath
+     * @return array
+     */
+    public function parseManuscriptReport(string $manuscriptPath): array
+    {
+        $manuscriptData = [];
+        $columnHeader = [
+            'Manuscript Number' => 'manuscript_number',
+            'Article Title' => 'article_title',
+            'Editorial Status Date' => 'editorial_status_date',
+            'Editorial Status' => 'editorial_status',
+        ];
```

Comment:
The column names for the `Manuscript` table are kind of hard-coded here. What happens if the EM query for the Manuscript report is changed to include new data? Maybe this function could be more generic to return an array of associative arrays - with each associative array representing a manuscript / author / review record. The keys of the associative record can be provided by the column names of the EM report.

Reply:
I was thinking inside `ManuscriptsWorker`, it will only contains methods  specifically for manuscript report, likes, `parseManusciptReport()` and `saveManuscript()`. While for the other type of EM report, would it be better to have different worker class, likes, `parseAuthorReport()`  in `AuthorWorker`, `parseReviewReport()` in `ReviewWorker`, etc .